cli/cmd/deploy/deploy.go

```diff
@@ -0,0 +1,397 @@
+				warn.Println(msg)
+				// NOTE : calling commands within commands has both pros and cons
+				// PRO : No duplicated code
+				// CON : Need to make sure that UX under sub command verifies with UX on this command
+				loginCmd := auth.LoginCmd(cfg)
+				loginCmd.SetContext(ctx)
+				if err := loginCmd.RunE(loginCmd, nil); err != nil {
+					exitWithFailure(err)
+				}
+
+				// set token in config
+				token, err := dotrill.GetAccessToken()
+				if err != nil {
+					return fmt.Errorf("could not parse access token from ~/.rill: %w", err)
+				}
+				cfg.AdminTokenDefault = token
+
+				// init admin client
+				adminClient, err = cmdutil.Client(cfg)
+				if err != nil {
+					return err
+				}
+			} else {
+				// switch is already part of login cmd so running this only when user is already logged in
+				defaultOrg, err := dotrill.GetDefaultOrg()
+				if err != nil {
+					return err
+				}
+
+				// init admin client
+				adminClient, err = cmdutil.Client(cfg)
+				if err != nil {
+					return err
+				}
+
+				multipleOrgs, err := multipleOrgs(adminClient)
+				if err != nil {
+					return fmt.Errorf("listing orgs failed with error %w", err)
+				}
+
+				if multipleOrgs {
+					msg := fmt.Sprintf("This project will be deployed under %s. Press Y to confirm and N to select a different org", defaultOrg)
+					if !cmdutil.ConfirmPrompt(msg, true) {
+						switchCmd := org.SwitchCmd(cfg)
+						switchCmd.SetContext(ctx)
+						if err := switchCmd.RunE(switchCmd, nil); err != nil {
+							exitWithFailure(err)
+						}
+					}
+				}
+			}
+
+			org, err := dotrill.GetDefaultOrg()
+			if err != nil {
+				return err
+			}
+
+			path, err := os.Getwd()
+			if err != nil {
+				return err
+			}
+
+			// verify current directory has rill project
+			if !hasRillProject(path) {
+				warn.Printf("\nCurrent path %s doesn't have a valid rill project. Please select correct path.\n", path)
+				warn.Printf("In case there is no valid rill project present, Please use `rill init` to create an empty rill project.\n\n")
+				// project path prompt
+				path = projectPathPrompt()
+			}
+
+			// verify project dir is a git repo with remote on github
+			githubURL, err := extractRemote(path)
+			if err != nil {
+				if errors.Is(err, gitutil.ErrGitRemoteNotFound) || errors.Is(err, git.ErrRepositoryNotExists) {
+					info.Print(githubSetupMsg)
+					return nil
+				}
+
+				return err
+			}
+
+			// Check for access to the Github URL
+			ghRes, err := project.VerifyAccess(ctx, adminClient, githubURL)
+			if err != nil {
+				return fmt.Errorf("failed to verify access to github repo, error = %w", err)
+			}
+
+			// We now have access to the Github repo
+			opts, err := projectParamPrompt(ctx, adminClient, org, githubURL, ghRes.DefaultBranch)
+			if err != nil {
+				return err
+			}
+
+			parsedVariables, err := variable.Parse(opts.Variables)
+			if err != nil {
+				return err
+			}
+
+			// fixing these for now
+			opts.dbDriver = "duckdb"
+			opts.slots = 2
```

Comment:
Use regular CLI flags for these like in `rill project connect`. We're removing `rill project connect`, so this command should support the same flexibility.

Reply:
I imagined building these as part of building non interactive mode. Since there is some overlap work here. At that time we can remove `rill project connect`.
We need to think on how to build non interactive mode vs interactive mode and some flags. Some considerations:
1. Should user be allowed to set anything via flag and still be in interactive mode
2. If user provides some input via cli flag, should we skip that question in interactive mode or use it just as default.
3. Or should we not allow user to set flags for inputs which are part of prompts in interactive mode.

Reply:
I think the behavior should be:

1. If a flag is provided, we use it
2. If a flag is not provided, some things we prompt for (like default branch), other things we just assume the default (like olap driver)
3. If `--non-interactive` is set, all required flags must be passed or it will error

But we don't need to built non-interactive now